src/NuGet.Core/NuGet.Packaging/PackageExtraction/ZipArchiveExtensions.cs

```diff
@@ -70,12 +70,11 @@ public static void UpdateFileTimeFromEntry(this ZipArchiveEntry entry, string fi
             var attr = File.GetAttributes(fileFullPath);
 
             if (!attr.HasFlag(FileAttributes.Directory) &&
-                entry.LastWriteTime.DateTime != DateTime.MinValue && // Ignore invalid times
-                entry.LastWriteTime.UtcDateTime <= DateTime.UtcNow) // Ignore future times
+                entry.LastWriteTime.DateTime != DateTime.MinValue) // Ignore invalid times
```

Comment:
why remove the future times check?

Reply:
If user's computer clock is off by few minutes/hours or if package owner really wanted to make package modified date set in the the future then we shouldn't be blocking it. I believe there is reason it's happening.

Reply:
Not sure I follow. 
The reason what's happening?